core/src/main/scala/akka/persistence/cassandra/cleanup/Cleanup.scala

```diff
@@ -40,15 +46,203 @@ final class Cleanup(system: ActorSystem, settings: CleanupSettings) {
 
   import settings._
   import system.dispatcher
+  private implicit val mat: Materializer = SystemMaterializer(system).materializer
 
   private val log = Logging(system, getClass)
   private val journal = Persistence(system).journalFor(journalPlugin)
-  private val snapshotStore =
-    if (snapshotPlugin != "" || system.settings.config.getString("akka.persistence.snapshot-store.plugin").nonEmpty)
-      Some(Persistence(system).snapshotStoreFor(snapshotPlugin))
-    else None
+  private val snapshotStore = Persistence(system).snapshotStoreFor(snapshotPlugin)
```

Comment:
Now it won't work if there was no snapshot store configured or what was that previous logic? Should it be lazily constructed so it is ok if it is never touched?

Reply:
ah yes missed lazy on that 👍